packages/roosterjs-content-model-plugins/lib/autoFormat/link/createLink.ts

```diff
@@ -1,11 +1,17 @@
 import { addLink, ChangeSource } from 'roosterjs-content-model-dom';
-import { formatTextSegmentBeforeSelectionMarker, matchLink } from 'roosterjs-content-model-api';
-import type { ContentModelLink, IEditor, LinkData } from 'roosterjs-content-model-types';
+import { formatTextSegmentBeforeSelectionMarker } from 'roosterjs-content-model-api';
+import { getLinkUrl } from './getLinkUrl';
+import type { ContentModelLink, IEditor } from 'roosterjs-content-model-types';
 
 /**
  * @internal
  */
-export function createLink(editor: IEditor) {
+export function createLink(
+    editor: IEditor,
```

Comment:
Let's use an option object instead of multiple parameters. You can create an option type for createLink, then let `AutoFormatOptions` extend from it.

```ts
interface AutoLinkOptions {
  autoLink?: boolean; 
 ...
}

interface AutoFormatOptions extends AutoLinkOptions {
  ...
}
```